src/components/ToggleSwitch/ToggleSwitch.tsx

```diff
@@ -0,0 +1,22 @@
+import { useToggle } from "hooks/useToggle";
+import { useEffect } from "react";
+import style from "./ToggleSwitch.module.scss";
+
+type Props = {
+    onToggle: (isOn: boolean) => void,
+}
+
+export function ToggleSwitch({ onToggle }: Props) {
+    const [isOn, flip] = useToggle();
+
+    const onClick = () => {
+        flip();
+        onToggle(isOn);
```

Comment:
You have to use the useEffect like in the ToggleButton because isOn will be the old value instead of the new one.